File: apps/backend/src/domain/value_objects/enums.py
from enum import Enum


class Role(str, Enum):
    NUTRITIONIST = "NUTRITIONIST"
    ADMIN = "ADMIN"


class Gender(str, Enum):
    MALE = "MALE"
    FEMALE = "FEMALE"
    OTHER = "OTHER"


class AppointmentType(str, Enum):
    INITIAL = "INITIAL"
    FOLLOWUP = "FOLLOWUP"
    RETURN = "RETURN"
    EMERGENCY = "EMERGENCY"
    ONLINE = "ONLINE"


class AppointmentStatus(str, Enum):
    SCHEDULED = "SCHEDULED"
    CONFIRMED = "CONFIRMED"
    IN_PROGRESS = "IN_PROGRESS"
    COMPLETED = "COMPLETED"
    CANCELLED = "CANCELLED"
    NO_SHOW = "NO_SHOW"


class PlanStatus(str, Enum):
    ACTIVE = "ACTIVE"
    COMPLETED = "COMPLETED"
    CANCELLED = "CANCELLED"


class TicketStatus(str, Enum):
    OPEN = "OPEN"
    IN_PROGRESS = "IN_PROGRESS"
    WAITING = "WAITING"
    RESOLVED = "RESOLVED"
    CLOSED = "CLOSED"


class Priority(str, Enum):
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    URGENT = "URGENT"


class MessageDirection(str, Enum):
    INBOUND = "INBOUND"
    OUTBOUND = "OUTBOUND"


class ConsentType(str, Enum):
    MARKETING = "MARKETING"
    ANALYTICS = "ANALYTICS"
    DATA_SHARING = "DATA_SHARING"
    WHATSAPP = "WHATSAPP"

File: packages/shared-types/src/enums.ts
export enum Role {
  NUTRITIONIST = "NUTRITIONIST",
  ADMIN = "ADMIN",
}

export enum Gender {
  MALE = "MALE",
  FEMALE = "FEMALE",
  OTHER = "OTHER",
}

export enum AppointmentType {
  INITIAL = "INITIAL",
  FOLLOWUP = "FOLLOWUP",
  RETURN = "RETURN",
  EMERGENCY = "EMERGENCY",
  ONLINE = "ONLINE",
}

export enum AppointmentStatus {
  SCHEDULED = "SCHEDULED",
  CONFIRMED = "CONFIRMED",
  IN_PROGRESS = "IN_PROGRESS",
  COMPLETED = "COMPLETED",
  CANCELLED = "CANCELLED",
  NO_SHOW = "NO_SHOW",
}

export enum PlanStatus {
  ACTIVE = "ACTIVE",
  COMPLETED = "COMPLETED",
  CANCELLED = "CANCELLED",
}

export enum TicketStatus {
  OPEN = "OPEN",
  IN_PROGRESS = "IN_PROGRESS",
  WAITING = "WAITING",
  RESOLVED = "RESOLVED",
  CLOSED = "CLOSED",
}

export enum Priority {
  LOW = "LOW",
  MEDIUM = "MEDIUM",
  HIGH = "HIGH",
  URGENT = "URGENT",
}

export enum MessageDirection {
  INBOUND = "INBOUND",
  OUTBOUND = "OUTBOUND",
}

export enum ConsentType {
  MARKETING = "MARKETING",
  ANALYTICS = "ANALYTICS",
  DATA_SHARING = "DATA_SHARING",
  WHATSAPP = "WHATSAPP",
}
